docs/adr/004-mithril-network-update-strategy.md

```diff
@@ -0,0 +1,72 @@
+---
+slug: 4
+title: |
+  4. Mithril Network Upgrade Strategy
+authors:
+- name: Mithril Team
+tags: [Draft]
+date: 2023-01-05
+---
+
+## Status
+
+**Draft**
+
+## Context
+
+When we will run Mithril on Mainnet there will be thousands of signers running altogether. Upgrading the version of the nodes has an impact as different versions of API, messages, signature may lead to loss of a significant part of the signers population over one epoch or more. In any case we must prevent a gap in the certificate chain while upgrading critical parts.
+
+We need to be able to keep enough of signer nodes and the aggregator able to work together in order to produce at least one certificate per epoch.
+
+Examples of such changes:
+
+ * change in the message structure
+ * change in the cryptographic algorithm
+ * change in communication channels
+
+## Decision
+
+In order to synchronize all nodes behavior transition, the Release Team will set up Eras that start at a defined Cardano Epoch and lasts until the next Era begins. When nodes detect an Era change, they switch from old to new behavior hence all transitioning at almost the same time.
```

Comment:
The team does not directly activate eras. They define at what Epoch it will be active, this definition is immutably written in the BlockChain. I am going to rephrase this.